### PR TITLE
[android] Fix my position arrow during navigation

### LINKED ISSUE #3441: [android] Position arrow overlaps with bottom bar in navigation mode.
Samsung Galaxy S20 FE (Snapdragon), latest beta:

Upd: looks like it happens only when device is locked

photo_2022-09-18 12 08 19

### PR DESCRIPTION
**Pull request by rtsisyk**
Closes #3441

Signed-off-by: Roman Tsisyk <roman@tsisyk.com>

### CODE REVIEW

File: android/jni/com/mapswithme/maps/Framework.cpp [hunk 1]
```diff
@@ -660,6 +660,11 @@ void Framework::SetupWidget(gui::EWidget widget, float x, float y, dp::Anchor an
   m_guiPositions[widget] = gui::Position(m2::PointF(x, y), anchor);
 }
 
+void Framework::UpdateMyPositionRoutingOffset(int offsetY)
+{
+  NativeFramework()->UpdateMyPositionRoutingOffset(false, offsetY);
```
Comment (biodranik) [inline android/jni/com/mapswithme/maps/Framework.cpp:665]: Why is this line can't be called directly from MapFragment.cpp? Why is another wrapper layer needed?

File: android/src/com/mapswithme/maps/MapFragment.java [hunk 2]
```diff
@@ -132,6 +132,8 @@ void setupBottomWidgetsOffset(int offsetY, int offsetX)
     int y = offsetY < 0 ? mBottomWidgetOffsetY : offsetY;
     setupRuler(y, x,true);
     setupAttribution(y, x, true);
+    Logger.w("TEST", "nativeSetupMyPosition: " + offsetY);
```
Comment (biodranik) [inline android/src/com/mapswithme/maps/MapFragment.java:135]: ?

File: android/src/com/mapswithme/maps/widget/menu/NavMenu.java [hunk 3]
```diff
@@ -50,10 +50,19 @@
 
   private int currentPeekHeight = 0;
 
-  public NavMenu(AppCompatActivity activity, NavMenuListener navMenuListener)
+
+  public interface OnMenuSizeChangedListener
```
Comment (biodranik) [inline android/src/com/mapswithme/maps/widget/menu/NavMenu.java:54]: Why listener? Isn't `mMapFragment.setupBottomWidgetsOffset` a proper place to update the C++ arrow position?

File: android/src/com/mapswithme/maps/MwmActivity.java [hunk 4]
```diff
@@ -478,7 +478,7 @@ private void initViews(boolean isLaunchByDeeplink)
       removeCurrentFragment(false);
     }
 
-    mNavigationController = new NavigationController(this, mMapButtonsController, v -> onSettingsOptionSelected());
+    mNavigationController = new NavigationController(this, mMapButtonsController, v -> onSettingsOptionSelected(), this::adjustBottomWidgets);
```
Comment (biodranik) [inline android/src/com/mapswithme/maps/MwmActivity.java:481]: How adjustBottomWidgets receives the argument from the new listener code below?

**Review by biodranik — CHANGES_REQUESTED** [review]

**Comment by rtsisyk** [comment]
This patch is not possible to rebase on the latest master.

**Pull request closed without merging**